[rm] update instructions for windows signing (#13591)
coq/coq: dev/doc/release-process.md

- [ ] Draft a release on GitHub.
- [ ] Sign the Windows and MacOS packages and upload them on GitHub.
  + The Windows packages must be signed by the Inria IT security service. They
    should be sent as a link to the binary (via [filesender](https://filesender.renater.fr) for example)
    together with its SHA256 hash in a signed e-mail to `dsi.securite` @ `inria.fr`
    putting `@maximedenes` in carbon copy.

>   + The MacOS packages should be signed by our own certificate, by sending them
>     to `@maximedenes`. A detailed step-by-step guide can be found [on the wiki](https://github.com/coq/coq/wiki/SigningReleases).

Reviewer: What about also fixing this?

```suggestion
  + The MacOS packages should be signed with our own certificate. A detailed step-by-step guide can be found [on the wiki](https://github.com/coq/coq/wiki/SigningReleases).
```
